### PR TITLE
HPCC-13981 Fix isses with filenames with leading numeric characters

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@richardkchapman - please review

File: dali/base/dautils.cpp [hunk 1]
```diff
@@ -908,7 +909,14 @@ StringBuffer &CDfsLogicalFileName::makeXPathLName(StringBuffer &lfnNodeName) con
                     c = toupper(*s);
                     // fall through
                 default:
-                    if (isalnum(c))
+                    if (first == s)
```
Comment (richardkchapman) [inline dali/base/dautils.cpp]: Not sure it really matters, but can someone now create a file a::_49b whose lock would collide with the file a::1b ? And possibly also clashes between files that start with chr(210) and ones that start with chr(21) then the digit 1 ?

Comment (richardkchapman) [inline dali/base/dautils.cpp]: You also seem to be using _ as a separator in place of ::  which increases the possible clashes.

File: dali/base/dautils.cpp [hunk 2]
```diff
@@ -894,6 +894,7 @@ StringBuffer &CDfsLogicalFileName::makeXPathLName(StringBuffer &lfnNodeName) con
                 first = false;
             else
                 lfnNodeName.append('_');
+            const char *first = s;
```
Comment (richardkchapman) [inline dali/base/dautils.cpp]: Hides the outer boolean variable of the same name...

File: dali/base/dautils.cpp [hunk 3]
```diff
@@ -894,6 +894,7 @@ StringBuffer &CDfsLogicalFileName::makeXPathLName(StringBuffer &lfnNodeName) con
                 first = false;
             else
                 lfnNodeName.append('_');
```
Comment (richardkchapman) [inline dali/base/dautils.cpp]: If you just made this unconditional it might be a more elegant fix?

**Comment by jakesmith** [comment]
@richardkchapman - please see changes.

**Pull request merged**